app/workers/asset_manager_attachment_set_uploaded_to_worker.rb

```diff
@@ -35,16 +35,27 @@ def perform(model_class, model_id, legacy_url_path)
       if attachment_data.path == legacy_url_path
         found = true
         attachment_data.uploaded_to_asset_manager!
```

Comment:
Definitely something to consider as part of the other epic around state syncing. 

Also on this one, I am very blurry as to the usefulness of the updates being called inside that method (given they also get called from the controller and other places) because the updates don't really include anything referring to this timestamp. Update calls sprinkled everywhere feels like a catch-all to cover all the points where the async nature of it all might crack.

@kornysietsma Tagging you so that this lives in your brain too - as mastermind of syncing.